client/components/gridicon/index.tsx

```diff
@@ -11,11 +10,19 @@ import {
 } from 'gridicons/dist/util/icons-offset';
 import spritePath from 'gridicons/svg-sprite/gridicons.svg';
 
-function needsOffset( name, icons ) {
+function needsOffset( name: string, icons: [string] ) {
 	return icons.indexOf( name ) >= 0;
 }
 
-const Gridicon = React.forwardRef( ( props, ref ) => {
+interface Props {
+	icon: string;
+	size?: number;
+	onClick?: ( event?: React.MouseEvent< SVGSVGElement, MouseEvent > ) => any;
+	className?: string;
+	[propName: string]: any;
```

Comment:
This opens up any props to be passed. I wonder if we can do something like this instead:

```ts
interface Props extends React.SVGAttributes< SVGSVGElement > {
	icon: string;
	size?: number;
	onClick?: React.MouseEventHandler< SVGSVGElement >;
	className?: string;
}
```

```suggestion
```

🤔 I'm on the fence about this, because while I believe this is technically more accurate (we pass additional props through to a React SVGElement), it may not be helpful.

Reply:
In my opinion, if we allow for it in runtime, we should allow for it at compile time. I think it would be odd to flag `<Gridicon icon="foo" unknown="prop" />` as invalid if it's something we want to work at runtime.

Reply:
I think you're right and this extension is probably a good way to go. The change should be:

```ts
interface Props extends React.SVGProps<SVGSVGElement> {
```

Reply:
I'll go for that, except in `MaterialIcon`, where `style` has a different meaning than in `<svg>` 😬

Reply:
> I'll go for that, except in `MaterialIcon`, where `style` has a different meaning than in `<svg>` 😬

It can just be overridden, right?

```ts
interface MaterialIconProps extends React.SVGProps<SVGSVGElement> {
  style: string;
}
```

Reply:
No, I don't think that's possible:

```
Interface 'Props' incorrectly extends interface 'SVGProps<SVGSVGElement>'.
  Types of property 'style' are incompatible.
    Type 'string | undefined' is not assignable to type 'CSSProperties | undefined'.
```

Reply:
Oh, I guess this works:

```ts
interface Props extends Omit< React.SVGProps< SVGSVGElement >, 'style' > {
	icon: string;
	style?: string;
	size?: number;
	onClick?: React.MouseEventHandler< SVGSVGElement >;
	className?: string;
	[propName: string]: any;
}
```

Blech 🤢

Reply:
> Oh, I guess this works:

Not the prettiest, but I guess that makes sense 🤷‍♂ I tried a few things and couldn't come up with anything better.

Without `[propName: string]: any;`, right? 😬

Reply:
> Without `[propName: string]: any;`, right? 😬

Yup, that's right. No need for `[propName: string]: any;` if we're basing this on everything the SVG element supports.